src/server.rs

```diff
@@ -590,6 +609,47 @@ mod tests {
         fs::remove_file(path_to_socket).unwrap();
     }
 
+    #[test]
+    fn test_wait_one_fd_connection() {
+        use std::os::unix::io::IntoRawFd;
+        let path_to_socket = "/tmp/test_socket_http_server1.sock";
+        fs::remove_file(path_to_socket).unwrap_or_default();
+        let socket_listener = UnixListener::bind(path_to_socket).unwrap();
+        let socket_fd = socket_listener.into_raw_fd();
+
+        let mut server = HttpServer::new_from_fd(socket_fd).unwrap();
+        server.start_server().unwrap();
+
+        // Test one incoming connection.
+        let mut socket = UnixStream::connect(path_to_socket).unwrap();
+        assert!(server.requests().unwrap().is_empty());
+
+        socket
+            .write_all(
+                b"PATCH /machine-config HTTP/1.1\r\n\
+                         Content-Length: 13\r\n\
+                         Content-Type: application/json\r\n\r\nwhatever body",
+            )
+            .unwrap();
+
+        let mut req_vec = server.requests().unwrap();
+        let server_request = req_vec.remove(0);
+
+        server
+            .respond(server_request.process(|_request| {
+                let mut response = Response::new(Version::Http11, StatusCode::OK);
+                let response_body = b"response body";
+                response.set_body(Body::new(response_body.to_vec()));
+                response
+            }))
+            .unwrap();
+        assert!(server.requests().unwrap().is_empty());
+
+        let mut buf: [u8; 1024] = [0; 1024];
+        assert!(socket.read(&mut buf[..]).unwrap() > 0);
```

Comment:
nit: you could also validate the contents

Reply:
Thanks for the feedback. That's a good point. I copied the other test without too much thought X(.

Updated to check the request body and the response body content.